### ISSUE TITLE
"Building CXX object" failed on debian arm64

### ISSUE THREAD

**Issue by HiDe-Techno-Tips** [xmrig/xmrig#2404]
**_Trying to build v6.12.1 on debian for arm64_**

Command `root@debian:~/xmrig/build# cmake..` completes successfully but,

Command `root@debian:~/xmrig/build# make` runs without error till `Scanning dependencies of target xmrig` and fails multiple times while  `Building CXX object`. Errors are found in multiple CXX header files saying `error: missing binary operator before token "("`

Heres the full [log](https://github.com/xmrig/xmrig/files/6527437/log.txt).

What do I do?

**Comment by Spudz76** [comment]
Seems related to Android SDK problem as seen here: https://github.com/abseil/abseil-cpp/issues/121

**Comment by HiDe-Techno-Tips** [comment]
Should I reinstall something?

**Comment by Spudz76** [comment]
Investigating.

Does `dpkg-query -l | grep gcc` all say `aarch64`?

**Comment by HiDe-Techno-Tips** [comment]
```
root@debian:~/xmrig/build# dpkg-query -l | grep gcc
ii  gcc                        4:10.2.1-1                     arm64        GNU C compiler
ii  gcc-10                     10.2.1-6                       arm64        GNU C compiler
ii  gcc-10-base:arm64          10.2.1-6                       arm64        GCC, the GNU Compiler Collection (base package)
ii  gcc-8-base:arm64           8.3.0-6                        arm64        GCC, the GNU Compiler Collection (base package)
ii  gcc-9-base:arm64           9.3.0-22                       arm64        GCC, the GNU Compiler Collection (base package)
ii  libgcc-10-dev:arm64        10.2.1-6                       arm64        GCC support library (development files)
ii  libgcc-s1:arm64            10.2.1-6                       arm64        GCC support library
ii  libgcc1:arm64              1:8.3.0-6                      arm64        GCC support library
```

Should i remove gcc-8-base and gcc-9-base?

**Comment by Spudz76** [comment]
Yes that might be something to do with it, but it does seem like the files it dislikes are properly the 10 version.

There are a lot of various similar problems on other software for arm64 there might be a workaround.

**Comment by cmxhost** [comment]
maybe you can try my binaries
https://github.com/cmxhost/xmrig/releases/tag/v6.12.1

**Comment by HiDe-Techno-Tips** [comment]
> maybe you can try my binaries
> https://github.com/cmxhost/xmrig/releases/tag/v6.12.1

**Will try that...**
@cmxhost I appreciate your arm64 binary release. Maybe the official @xmrig should also add one.

**Comment by HiDe-Techno-Tips** [comment]
**_UPDATE_**

Compiling on Alpine for arm(aarch64):
Fails while executing `make` yeilds the following [logs](https://github.com/xmrig/xmrig/files/6528224/log.txt).

**Comment by cmxhost** [comment]
> **_UPDATE_**
> 
> Compiling on Alpine for arm(aarch64):
> Fails while executing `make` yeilds the following [logs](https://github.com/xmrig/xmrig/files/6528224/log.txt).

I don't know how you done in cmake command,  for me works fine

**Comment by HiDe-Techno-Tips** [comment]
why does asm/hwcap.h not exist for me? How to add it?

**Comment by cmxhost** [comment]
> why does asm/hwcap.h not exist for me? How to add it?

what devices you using ?

**Comment by HiDe-Techno-Tips** [comment]
right now alpine on aarch64 qemu vm

**Comment by cmxhost** [comment]
![Uploading Screenshot_20210523_224139.jpg…]()

**Comment by cmxhost** [comment]
![Screenshot_20210523_224139](https://user-images.githubusercontent.com/82912427/119265320-d7c23c80-bc18-11eb-84aa-6c9eb3206dde.jpg)
![Screenshot_20210523_224200](https://user-images.githubusercontent.com/82912427/119265326-dc86f080-bc18-11eb-9d27-860cc105b928.jpg)
![Screenshot_20210523_224324](https://user-images.githubusercontent.com/82912427/119265329-e01a7780-bc18-11eb-8b29-3dc38fbabae7.jpg)


All works fine

**Comment by HiDe-Techno-Tips** [comment]
Does your binary run directly on termux?

**Comment by cmxhost** [comment]
> Does your binary run directly on termux?

yes i just run on pi 4 and termux

**Comment by HiDe-Techno-Tips** [comment]
when i try to run in termux, it says `no such file or directory found`, even when xmrig is present in working directory.

**Comment by cmxhost** [comment]
> when i try to run in termux, it says `no such file or directory found`, even when xmrig is present in working directory.

if you run the binaries on termux debian or ubuntu no problem for sure

**Comment by thosoo** [comment]
> **_UPDATE_**
> 
> Compiling on Alpine for arm(aarch64):
> Fails while executing `make` yeilds the following [logs](https://github.com/xmrig/xmrig/files/6528224/log.txt).

```
localhost:~/xmrig/build# cmake .. -DXMRIG_DEPS=scripts/deps
```

what do you have in scripts/deps

**Comment by Spudz76** [comment]
```
localhost:~/xmrig/build# cmake .. -DXMRIG_DEPS=scripts/deps
```
That should be `../scripts/deps`
